### PR TITLE
Add API to create/amend dividends

### PR DESCRIPTION
**Pull request by Jxr227**

### CODE REVIEW

File: app/connectors/httpParsers/CreateOrAmendDividendsHttpParser.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+package connectors.httpParsers
+
+import play.api.http.Status.{NOT_FOUND, OK, SERVICE_UNAVAILABLE}
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+
+object CreateOrAmendDividendsHttpParser {
+  type CreateOrAmendDividendsResponse = Either[CreateOrAmendDividendsException, Boolean]
+
+  implicit object SubmittedDividendsHttpReads extends HttpReads[CreateOrAmendDividendsResponse] {
```
Comment (dnyuar) [inline app/connectors/httpParsers/CreateOrAmendDividendsHttpParser.scala:25]: Should be 'CreateOrAmendDividendsHttpReads' not 'SubmittedDividendsHttpReads'

File: app/controllers/CreateOrAmendDividendsController.scala [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ * limitations under the License.
+ */
+
+package controllers
+
+import connectors.httpParsers.CreateOrAmendDividendsHttpParser.CreateOrAmendDividendsResponse
+import controllers.predicates.AuthorisedAction
+import javax.inject.Inject
+import models.CreateOrAmendDividendsModel
+import play.api.libs.json.{JsResult, JsSuccess}
```
Comment (dnyuar) [inline app/controllers/CreateOrAmendDividendsController.scala:23]: JsResult is an unused import

File: app/controllers/CreateOrAmendDividendsController.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  def createOrAmendDividends(nino: String, taxYear:Int, mtditid: String): Action[AnyContent] = authorisedAction.async(mtditid) { implicit user =>
+    user.request.body.asJson.map(_.validate[CreateOrAmendDividendsModel]) match {
+      case Some(JsSuccess(model, _)) => responseHandler(createOrAmendDividendsService.createOrAmendDividends(nino, taxYear, model))
+      case _ => Future.successful(BadRequest)
+    }
+  }
+
+  def responseHandler(serviceResponse: Future[CreateOrAmendDividendsResponse]): Future[Result] ={
+    serviceResponse.map {
+      case Right(responseModel) => NoContent
```
Comment (dnyuar) [inline app/controllers/CreateOrAmendDividendsController.scala:44]: 'responseModel' and 'error' are declarations which are not used but are they there to explain what the cases do?

File: app/models/CreateOrAmendDividendsModel.scala [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+import play.api.libs.json.{JsError, JsResult, JsValue, Json, Reads, Writes}
```
Comment (dnyuar) [inline app/models/CreateOrAmendDividendsModel.scala:19]: JsResult is an unused import

File: it/connectors/CreateOrAmendDividendsConnectorSpec.scala [hunk 5]
```diff
@@ -0,0 +1,67 @@
+
+package connectors
+
+import connectors.httpParsers.CreateOrAmendDividendsHttpParser.{CreateOrAmendDividendsNotFoundException, CreateOrAmendDividendsServiceUnavailableException, CreateOrAmendDividendsUnhandledException}
+import connectors.httpParsers.SubmittedDividendsHttpParser._
```
Comment (dnyuar) [inline it/connectors/CreateOrAmendDividendsConnectorSpec.scala:5]: unused imports (SubmittedDividendsHttpParser & SubmittedDividendsModel)

**Review by dnyuar — CHANGES_REQUESTED** [review]

File: app/controllers/CreateOrAmendDividendsController.scala (hunk 3 above)
Comment (Jxr227) [reply app/controllers/CreateOrAmendDividendsController.scala:44]: Yeah it's more just explicit matching so we've got slightly more descriptive code

Comment (Jxr227) [reply app/controllers/CreateOrAmendDividendsController.scala:44]: (And as we iterate error cases the errorModel will be used in future)

**Pull request merged**